pkg/controller/daemon/daemon_controller.go

```diff
@@ -519,6 +526,67 @@ func (dsc *DaemonSetsController) updatePod(old, cur interface{}) {
 	}
 }
 
+// listSuspendedDaemonPods lists the Daemon pods that 'want to run, but should not schedule'
+// for the node.
+func (dsc *DaemonSetsController) listSuspendedDaemonPods(node string) (dss []string) {
+	dsc.suspendedDaemonPodsMutex.Lock()
+	defer dsc.suspendedDaemonPodsMutex.Unlock()
+
+	if _, found := dsc.suspendedDaemonPods[node]; !found {
+		return nil
+	}
+
+	for k := range dsc.suspendedDaemonPods[node] {
+		dss = append(dss, k)
+	}
+	return
+}
+
+// requeueSuspendedDaemonPods enqueues all DaemonSets which has pods that 'want to run,
+// but should not schedule' for the node; so DaemonSetController will sync up them again.
+func (dsc *DaemonSetsController) requeueSuspendedDaemonPods(node string) {
+	dss := dsc.listSuspendedDaemonPods(node)
+	for _, dsKey := range dss {
+		if ns, name, err := cache.SplitMetaNamespaceKey(dsKey); err != nil {
+			glog.Errorf("Failed to get DaemonSet's namespace and name from %s: %v", dsKey, err)
+			continue
+		} else if ds, err := dsc.dsLister.DaemonSets(ns).Get(name); err != nil {
+			glog.Errorf("Failed to get DaemonSet %s/%s: %v", ns, name, err)
+			continue
+		} else {
+			dsc.enqueueDaemonSet(ds)
```

Comment:
Pod deletions are not unsual, right? Application upgrades, node upgrades, rescheduling, garbage collection, all involve pod deletions.